### PR TITLE
Document ipc configuration file

### PR DESCRIPTION
**Pull request by SatpalSandhu61**
Add missing documentation to describe optional use of a TOML file to define IPC socket connection to transaction manager.
Along with new fields for configuring IPC socket timeouts, which were added in the following PR: https://github.com/ConsenSys/quorum/pull/1100

### CODE REVIEW

**Comment by NicolasMassart** [comment]
It seems important enough to be more than a note. You may create two headers `## Direct IPC connection` and `## Use IPC connection configuration file` and develop both ways in these sections.

File: docs/Concepts/Privacy/PrivateTransactionManager.md [hunk 1]
```diff
@@ -6,14 +6,45 @@ private transactions.
...
 of encrypted private transaction data between recipients of a private transaction.
 
+The Private Transaction Manager has two distinct components:
+
+* Transaction Manager
+* Enclave
+
 To enable private transactions, use the `PRIVATE_CONFIG` environment variable when starting a GoQuorum
-node to provide the node with the path to the Privacy Manager's `.ipc` socket, e.g.:
+node to provide the path to the Privacy Manager's `.ipc` socket. There are two ways to do this, which are described below.
```
Comment (NicolasMassart) [inline docs/Concepts/Privacy/PrivateTransactionManager.md:15]:
```suggestion
node to provide the path to the Privacy Manager's `.ipc` socket.

Use the `PRIVATE_CONFIG` environment variable either with a direct path to the IPC socket or with a path to a configuration file as explained in the following sections. 
```

File: docs/Concepts/Privacy/PrivateTransactionManager.md [hunk 2]
```diff
@@ -6,14 +6,45 @@ private transactions.
...
+
+## Direct IPC connection configuration
+
+This is the normal method for specifying the IPC socket:
```
Comment (NicolasMassart) [inline docs/Concepts/Privacy/PrivateTransactionManager.md:19]:
```suggestion
This is the basic method for specifying the IPC socket:
```

File: docs/Concepts/Privacy/PrivateTransactionManager.md [hunk 3]
```diff
@@ -6,14 +6,45 @@ private transactions.
...
 ```bash
 export PRIVATE_CONFIG=path/to/tm.ipc
 ```
 
-The Private Transaction Manager has two components:
+## Using an IPC connection configuration file
 
-* Transaction Manager
-* Enclave
+Alternatively, a TOML configuration file can be used to specify the details of the IPC socket:
```
Comment (NicolasMassart) [inline docs/Concepts/Privacy/PrivateTransactionManager.md:27]:
```suggestion
Use a TOML configuration file can to specify more options of the IPC socket:
```

File: docs/Concepts/Privacy/PrivateTransactionManager.md [hunk 4]
```diff
@@ -6,14 +6,45 @@ private transactions.
...
+
+```bash
+export PRIVATE_CONFIG=path/to/ipc-config-file.toml
+```
+
+Using this method allows the configuration of timeouts for the ipc socket. Here is an example of the TOML file:
```
Comment (NicolasMassart) [inline docs/Concepts/Privacy/PrivateTransactionManager.md:33]:
```suggestion
Configuration file provides the ability to configure timeouts for the ipc socket.
```

File: docs/Concepts/Privacy/PrivateTransactionManager.md [hunk 5]
```diff
@@ -6,14 +6,47 @@ private transactions.
...
+The Private Transaction Manager has two distinct components:
+
+* Transaction Manager
+* Enclave
+
 To enable private transactions, use the `PRIVATE_CONFIG` environment variable when starting a GoQuorum
-node to provide the node with the path to the Privacy Manager's `.ipc` socket, e.g.:
+node to provide the path to the Privacy Manager's `.ipc` socket.
+
+Use the `PRIVATE_CONFIG` environment variable either with a direct path to the IPC socket or with a path to a configuration file as explained in the following sections.
```
Comment (SatpalSandhu61) [inline docs/Concepts/Privacy/PrivateTransactionManager.md]: @NicolasMassart Your amendment is now duplicating the text "Use the `PRIVATE_CONFIG` environment variable'".

File: docs/Concepts/Privacy/PrivateTransactionManager.md [hunk 6]
```diff
@@ -6,14 +6,47 @@ private transactions.
...
+
 To enable private transactions, use the `PRIVATE_CONFIG` environment variable when starting a GoQuorum
-node to provide the node with the path to the Privacy Manager's `.ipc` socket, e.g.:
+node to provide the path to the Privacy Manager's `.ipc` socket.
+
+Use the `PRIVATE_CONFIG` environment variable either with a direct path to the IPC socket or with a path to a configuration file as explained in the following sections. 
+
+## Direct IPC connection configuration
+
+This is the basic method for specifying the IPC socket:
```
Comment (SatpalSandhu61) [inline docs/Concepts/Privacy/PrivateTransactionManager.md:22]: I would have preferred to leave this saying that this is the "normal" method as we want to discourage folks from using the more complex method of having a TOML file unless it's really necessary.

File: docs/Concepts/Privacy/PrivateTransactionManager.md [hunk 7]
```diff
@@ -6,14 +6,47 @@ private transactions.
...
 ```bash
 export PRIVATE_CONFIG=path/to/tm.ipc
 ```
 
-The Private Transaction Manager has two components:
+## Using an IPC connection configuration file
 
-* Transaction Manager
-* Enclave
+Use a TOML configuration file can to specify more options of the IPC socket:
```
Comment (SatpalSandhu61) [inline docs/Concepts/Privacy/PrivateTransactionManager.md]: Mistake in the grammar - I think it should read:
"Use a TOML configuration file to specify more options for the IPC socket:"

File: docs/Concepts/Privacy/PrivateTransactionManager.md (hunk 6 above)
Comment (NicolasMassart) [reply docs/Concepts/Privacy/PrivateTransactionManager.md:22]: Why? Basic means you fulfil basic needs which is the usual use case, but normal implies that the other one is not normal and thus a bit hacky.

File: docs/Concepts/Privacy/PrivateTransactionManager.md (hunk 7 above)
Comment (NicolasMassart) [reply docs/Concepts/Privacy/PrivateTransactionManager.md]: Yes will fix that, I'm making changes right now, ignore for the next 20 mins ;)

File: docs/Concepts/Privacy/PrivateTransactionManager.md (hunk 5 above)
Comment (NicolasMassart) [reply docs/Concepts/Privacy/PrivateTransactionManager.md]: We can repeat this kind of things if it makes the text easier to read. Sometimes using references like "this" or "previous" is not clear enough for non english speakers of when translated using Google Translate (lof of Chinese users using Translate for instance).

**Comment by NicolasMassart** [comment]
@SatpalSandhu61 let me know what you think of the changes I made in https://pegasys-docgoquorum--47.com.readthedocs.build/en/47/Concepts/Privacy/PrivateTransactionManager/
Thanks.

**Review by NicolasMassart — APPROVED** [review]

**Pull request merged**